### ISSUE TITLE
💅 useOptionalChain does not flag negation + comparison form (!foo || foo.bar !== "x")

### ISSUE THREAD

**Issue by cpirich** [biomejs/biome#10244]
### Environment information

```
@biomejs/biome 2.4.13 (also reproduces on latest 2.4.14)
```

### Rule name

`lint/complexity/useOptionalChain`

### Description

Issue #7211 / PR #9567 added support for the negated form `!foo || !foo.bar` → `!foo?.bar`. However, the closely-related De Morgan-equivalent form where the right-hand side of the `||` is a *comparison expression* on the same object is still not flagged:

```ts
// Flagged (good — handled by PR #9567)
if (!foo || !foo.bar) { ... }

// NOT flagged (gap)
if (!foo || foo.bar !== "x") { ... }
if (!foo || foo.bar === "x") { ... }
if (!foo || foo.bar > 0)     { ... }
```

The second form is semantically equivalent to `foo?.bar !== "x"` because `undefined !== "x"` is `true`, which short-circuits the same way `!foo` does in the original. ESLint's `@typescript-eslint/prefer-optional-chain` rule already handles this case, which makes the gap noticeable for projects migrating from ESLint to Biome.

### Reproduction

```ts
declare const refreshed: { status: string; accessToken: string | null } | null;

// Case A: !foo || !foo.bar — flagged ✓
if (!refreshed || !refreshed.accessToken) {}

// Case B: !foo || foo.bar !== "x" — NOT flagged ✗
if (!refreshed || refreshed.status !== "connected") {}

// Case C: !foo || !foo.bar || !foo.baz — flagged ✓ (chained negation)
if (!refreshed || !refreshed.accessToken || !refreshed.status) {}
```

Only Case A and Case C are reported on Biome 2.4.13. Case B should also be reportable as `refreshed?.status !== "connected"`.

### Expected result

Case B is flagged with a fix suggestion: `refreshed?.status !== "connected"`.

The same logic applies to `===`, `<`, `>`, `<=`, `>=`, `!=`, etc. — any binary comparison whose result on `undefined` matches the truthiness of `!foo`.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by cpirich** [comment]
Thanks — and apologies, my original report overreached. I want to retract part of it and narrow the bug to the case that's actually equivalent.

**Reproduction (Biome 2.4.14):** [open in playground](https://biomejs.dev/playground/?files.repro.ts=LwAvACAAQgB1AGcAIAByAGUAcAByAG8AIABmAG8AcgAgAGIAaQBvAG0AZQBqAHMALwBiAGkAbwBtAGUAIwAxADAAMgA0ADQACgAvAC8AIAB1AHMAZQBPAHAAdABpAG8AbgBhAGwAQwBoAGEAaQBuACAAZABvAGUAcwAgAG4AbwB0ACAAZgBsAGEAZwAgAHQAaABlACAAbgBlAGcAYQB0AGkAbwBuAC0AdwBpAHQAaAAtAGMAbwBtAHAAYQByAGkAcwBvAG4AIABmAG8AcgBtAAoACgBkAGUAYwBsAGEAcgBlACAAYwBvAG4AcwB0ACAAcgBlAGYAcgBlAHMAaABlAGQAOgAKACAAIAB8ACAAewAgAHMAdABhAHQAdQBzADoAIABzAHQAcgBpAG4AZwA7ACAAYQBjAGMAZQBzAHMAVABvAGsAZQBuADoAIABzAHQAcgBpAG4AZwAgAHwAIABuAHUAbABsACAAfQAKACAAIAB8ACAAbgB1AGwAbAA7AAoACgAvAC8AIABDAEEAUwBFACAAQQAgABQgIABmAGwAYQBnAGcAZQBkACAAYwBvAHIAcgBlAGMAdABsAHkAIAAoAGIAYQBzAGUAbABpAG4AZQAsACAAZgBvAHIAIABjAG8AbQBwAGEAcgBpAHMAbwBuACkACgAvAC8AIAAgACAAcwB1AGcAZwBlAHMAdABlAGQAIABmAGkAeAA6ACAAIQByAGUAZgByAGUAcwBoAGUAZAA%2FAC4AYQBjAGMAZQBzAHMAVABvAGsAZQBuAAoAaQBmACAAKAAhAHIAZQBmAHIAZQBzAGgAZQBkACAAfAB8ACAAIQByAGUAZgByAGUAcwBoAGUAZAAuAGEAYwBjAGUAcwBzAFQAbwBrAGUAbgApACAAewAKACAAIABjAG8AbgBzAG8AbABlAC4AbABvAGcAKAAiAGEAIgApADsACgB9AAoACgAvAC8AIABDAEEAUwBFACAAQgAgABQgIABOAE8AVAAgAGYAbABhAGcAZwBlAGQAIAAoAHQAaABlACAAYgB1AGcAKQAKAC8ALwAgACAAIABzAGUAbQBhAG4AdABpAGMAYQBsAGwAeQAgAGUAcQB1AGkAdgBhAGwAZQBuAHQAIAB0AG8AOgAgACAAcgBlAGYAcgBlAHMAaABlAGQAPwAuAHMAdABhAHQAdQBzACAAIQA9AD0AIAAiAGMAbwBuAG4AZQBjAHQAZQBkACIACgAvAC8AIAAgACAAYgBlAGMAYQB1AHMAZQAgAHcAaABlAG4AIAByAGUAZgByAGUAcwBoAGUAZAAgAGkAcwAgAG4AdQBsAGwAOgAKAC8ALwAgACAAIAAgACAAIQByAGUAZgByAGUAcwBoAGUAZAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAPQA9AD0AIAB0AHIAdQBlAAoALwAvACAAIAAgACAAIAByAGUAZgByAGUAcwBoAGUAZAA%2FAC4AcwB0AGEAdAB1AHMAIAAhAD0APQAgACIAYwBvAG4AbgBlAGMAdABlAGQAIgAgAD0APQA9ACAAKAB1AG4AZABlAGYAaQBuAGUAZAAgACEAPQA9ACAAIgBjAG8AbgBuAGUAYwB0AGUAZAAiACkAIAA9AD0APQAgAHQAcgB1AGUACgAvAC8AIAAgACAAKABhAGwAcwBvACAAYQBwAHAAbABpAGUAcwAgAHQAbwAgACEAPQApAAoAaQBmACAAKAAhAHIAZQBmAHIAZQBzAGgAZQBkACAAfAB8ACAAcgBlAGYAcgBlAHMAaABlAGQALgBzAHQAYQB0AHUAcwAgACEAPQA9ACAAIgBjAG8AbgBuAGUAYwB0AGUAZAAiACkAIAB7AAoAIAAgAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoACIAYgAiACkAOwAKAH0ACgA%3D)

```ts
declare const refreshed:
  | { status: string; accessToken: string | null }
  | null;

// CASE A — flagged ✓
if (!refreshed || !refreshed.accessToken) {}

// CASE B — NOT flagged (the bug)
// equivalent to: refreshed?.status !== "connected"
if (!refreshed || refreshed.status !== "connected") {}
```

Local CLI confirms this on `@biomejs/biome@2.4.14`: only Case A is reported.

**Correction to the original report:** I claimed "the same logic applies to `===`, `<`, `>`, `<=`, `>=`, `!=`, etc." That's wrong. Only operators whose result against `undefined` is truthy are safe to convert. Concretely:

| Original | Optional-chain | Equivalent when `foo` is null? |
| --- | --- | --- |
| `!foo \|\| foo.bar !== "x"` | `foo?.bar !== "x"` | ✅ both `true` (`undefined !== "x"`) |
| `!foo \|\| foo.bar != null` | `foo?.bar != null` | ✅ both `true` |
| `!foo \|\| foo.bar === "x"` | `foo?.bar === "x"` | ❌ `true` vs. `false` |
| `!foo \|\| foo.bar > 0` | `foo?.bar > 0` | ❌ `true` vs. `false` |

So the actionable bug is just **`!==` and `!=`** (with the usual caveat that `!= null` collapses both `null` and `undefined`). This matches what `@typescript-eslint/prefer-optional-chain` reports.
